tests/test_httpstatus.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8
+
+import ddt
+
+import falcon.testing as testing
+import falcon
+from falcon.http_status import HTTPStatus
+
+
+def before_hook(req, resp, params):
+    raise HTTPStatus(falcon.HTTP_200,
+                     headers={"X-Failed": "False"},
+                     body="Pass")
+
+
+def after_hook(req, resp, resource):
+    resp.status = falcon.HTTP_200
+    resp.set_header("X-Failed", "False")
+    resp.body = "Pass"
+
+
+class TestStatusResource:
+
+    @falcon.before(before_hook)
+    def on_get(self, req, resp):
+        resp.status = falcon.HTTP_500
+        resp.set_header("X-Failed", "True")
+        resp.body = "Fail"
+
+    def on_post(self, req, resp):
+        resp.status = falcon.HTTP_500
+        resp.set_header("X-Failed", "True")
+        resp.body = "Fail"
+
+        raise HTTPStatus(falcon.HTTP_200,
+                         headers={"X-Failed": "False"},
+                         body="Pass")
+
+    @falcon.after(after_hook)
+    def on_put(self, req, resp):
+        raise HTTPStatus(falcon.HTTP_500,
+                         headers={"X-Failed": "True"},
+                         body="Fail")
+
+
+@ddt.ddt
```

Comment:
Currently, it doesn't look like you're using ddt in this class. I would either use it to reduce the duplication between the 3 test methods or remove it.